### ISSUE TITLE
Create a more geneic ˋeventˋ class

### ISSUE THREAD

**Issue by Frontrider** [CottonMC/FunctionAPI#6]
**Is your feature request related to a problem? Please describe.**
As the engine supports event handlers other than commands (java included, javascript WIP) we should use something more generic than the ˋCommandSourceˋ.

**Describe the solution you'd like**
 A new ˋEventˋ class that can better accomodate new script engines.

Requested fields:

- position(vector3)
- block (optional)
- sourceEntity(optional)
- Target for sending feedback
- members: an array containin additional entities that are involved, for example the other animal during breeding, or the true source of damage.

Additionally:
It should also have an ˋexecuteˋ style command to switch into these members. The command context should not change, unless an ˋasˋ subcommand was called.

**Describe alternatives you've considered**
The current solution is to use the built-in ˋServerCommandSourceˋ class, as this is what commands use.

**Comment by LemmaEOF** [comment]
We'd definitely need some way to either convert to a ServerCommandSource or wrap a ServerCommandSource in our EventSource.

**Comment by Frontrider** [comment]
I'd probably mix into the `ServerCommandSource` to contain the required fields, the convert our `Event` to it. I'd like to remove minecraft classes from the core of the system.

**Comment by Frontrider** [comment]
This is the current way of creating command sources:
https://github.com/CottonMC/FunctionAPI/blob/master/src/main/java/io/github/cottonmc/functionapi/ServerCommandSourceFactory.java

**Comment by LemmaEOF** [comment]
Cotton Scripting [has its own command-source-creating methods](https://github.com/CottonMC/CottonScripting/blob/master/src/main/java/io/github/cottonmc/cotton_scripting/api/ScriptTools.java#L91-L178).

**Comment by Frontrider** [comment]
I wanted to avoid parsing strings in the middle of my code. And I already have all the members at hand. No need to search for them again.

**Comment by LemmaEOF** [comment]
Yeah yeah, these were meant for calling from JS code in obf environment.

**Comment by Frontrider** [comment]
Would be great if it got integrated into this class:
https://github.com/CottonMC/FunctionAPI/blob/master/src/main/java/io/github/cottonmc/functionapi/events/runners/script/api/ScriptApiObject.java
